### ISSUE TITLE
loss = bce + beta * kl

### ISSUE THREAD

**Issue by tumis1946** [riannevdberg/sylvester-flows#5]
hello Rianne:
Thanks very much. I am a bit confused with line 44 in loss.py : loss = bce + beta * kl. Based on equation 3 in Tomczak's paper (Improving Variational Auto-Encoder Using Householder Flows), shouldn't "loss = bce - beta * kl "? Also, why use -ELBO instead of ELBO when reporting your metrics?
Thanks

**Comment by riannevdberg** [comment]
Hi,

>I am a bit confused with line 44 in loss.py : loss = bce + beta * kl. Based on equation 3 in Tomczak's paper (Improving Variational Auto-Encoder Using Householder Flows), shouldn't "loss = bce - beta * kl "?

We want to maximize the log likelihood (intractable), and the ELBO is a lower bound to the log likelihood so maximizing the ELBO is what we want to do. This is equivalent to minimizing the -ELBO. -ELBO is the negative of the expression that is in eq 3 of Tomczak's paper. So loss = -ELBO = - E_q(z|x)[p(x|z)] + KL(q(z|x)|p(z)). Note the binary cross entropy (bce) has a minus sign in its definition: bce(q||p) = - E_q(z|x)[p(x|z)].

>Also, why use -ELBO instead of ELBO when reporting your metrics?

This is just a choice that doesn't really matter, several papers in the literature report -ELBO. Either you report ELBO and higher is better, or you report -ELBO and lower is better.

I hope this helps.

**Comment by tumis1946** [comment]
Thanks very much for the clarification. I greatly appreciate your help.
 The caption of Figure 2 in your paper has the following: "our inference network produces input-dependent flow parameters through a hypernetwork". Does this imply that the system will not learn if the hypernetwork was excluded? Or, even if it learns, the flow parameters will not be input-dependent?

**Comment by riannevdberg** [comment]
If you freeze the hypernetwork then the flow parameters will still be input dependent but not learned. If you remove the hypernetwork and instead replace the flow parameters directly with learnable parameters then they will be input independent. In that last case the only dependence of the parameters that determine the approximate posterior on the input is through the mean and log variance of the base distribution q(z_0|x) of the approximate posterior.

**Comment by tumis1946** [comment]
Thanks very much.
